certipy/test/test_certipy.py

```diff
@@ -373,10 +374,16 @@ def test_certs():
         ) as server:
             # Execute/Verify
             url = f"https://{server.host}:{server.port}"
-
             # Fails without specifying a CA for verification
-            with pytest.raises(SSLError):
-                requests.get(url)
+            with pytest.raises(URLError, match="SSL"):
+                with urlopen(url):
+                    pass
+
+            ssl_context = ssl.create_default_context(ssl.Purpose.SERVER_AUTH, cafile=ca_record["files"]["cert"])
+            ssl_context.verify_mode = ssl.CERT_REQUIRED
+            ssl_context.load_default_certs()
+            ssl_context.load_cert_chain(ca_record["files"]["cert"], ca_record["files"]["key"])
 
             # Succeeds when supplying the CA cert
-            requests.get(url, verify=ca_record["files"]["cert"])
```

Comment:
btw, the reason I switched from requests to urlopen here is that requests does _not_ use the default ssl config and this test still passed, even on 3.13. Only http libraries that use more default ssl setup (urllib, httpx, tornado, etc.) see this.

Reply:
Interesting, I figured that might have been why you did that. So it seems the default ssl setup is more strict than what `requests` is enforcing. Thanks for letting me know.

Reply:
Yeah, requests predates a lot of the standard library SSL stuff and is pretty hardcore about not changing default behavior no matter what, so changes in the standard library often don't affect requests.